### PR TITLE
parser: fix parsing of big integer values

### PR DESCRIPTION
**Pull request by belochub**
Implement additional function for parsing integer values which
cannot be parsed using standard library functions.

Refs: https://github.com/metarhia/jstp/issues/175

### CODE REVIEW

**Comment by belochub** [comment]
@aqrln, ping.

File: src/jsrs_parser.cc [hunk 1]
```diff
@@ -375,12 +378,48 @@ Local<Value> ParseIntegerNumber(Isolate*    isolate,
...
+  char base_alpha_count = base > 10 ? base - 10 : 0;
+  for (size_t i = 0; i < *size; i++) {
+    current_digit = toupper(begin[i]);
+    if ((current_digit < '0' || current_digit >= '0' + base_digit_count) &&
+        (current_digit < 'A' || current_digit >= 'A' + base_alpha_count)) {
+      *size = i;
+      break;
+    }
+    current_digit_value = current_digit <= '9' ? current_digit - '0' :
+      current_digit - 'A' + 10;
```
Comment (aqrln) [inline src/jsrs_parser.cc]: I believe this should be indented with four spaces.

Comment (belochub) [inline src/jsrs_parser.cc]: @aqrln, `cpplint` didn't complain about it, so I left it like that.
What do you say about moving it to the right, right after the question sign?

Comment (aqrln) [inline src/jsrs_parser.cc]: @belochub sounds good to me

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/jstp/commit/417ef60bdde3e9a68349399f6e7ef2b1e7bb4f1a.

**Pull request closed without merging**